src/game/client/gui/gamewindow.h

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+
+#include "always.h"
+#include "coord.h"
+#include "gamelogic.h"
+#include "mempoolobj.h"
+#include "unicodestring.h"
+#include "wininstancedata.h"
+
+class GameFont;
+
+// TODO confirm
+enum WindowMsgHandledType
+{
+    MSG_IGNORED,
+    MSG_HANDLED,
+};
+
+typedef WindowMsgHandledType(__cdecl *window_callback_func)(
+    GameWindow *window, unsigned int message, unsigned int data_1, unsigned int data_2);
+
+typedef void(__cdecl *window_draw_func)(GameWindow *window, WinInstanceData *instance);
```

Comment:
I don't think there is an entry in the style guide for function pointers but I would expect them to follow either the casing of classes or the casing of functions and not the style of variables.

Reply:
I'm not sure if I've done any function pointers as of yet, I've probably done it as one of the type styles, callback_func_t or CallbackFunc. I might be inclined to drop the "func" when callback is also mentioned as well, callback implies it. So shall we go with WindowCallback and WindowDrawFunc?